File: Backend/BuildingManagerDomain/Entities/Invitation.cs
using System;
using BuildingManagerDomain.Enums;

namespace BuildingManagerDomain.Entities
{
    public class Invitation
    {
        public Guid Id { get; set; }
        public string Email { get; set; }
        public string Name { get; set; }
        public long Deadline { get; set; }
        public InvitationStatus Status { get; set; }

        public override bool Equals(object obj)
        {
            Invitation invitation = (Invitation)obj;
            return Name == invitation.Name && Email == invitation.Email && Deadline == invitation.Deadline && Status == invitation.Status;
        }
    }
}

File: Frontend/src/app/models/Invitation.ts
import { UserRole } from "../enums/UserRole";

export interface Invitation {
    id?: string;
    email: string;
    name: string;
    deadline: number;
    role: UserRole;
}

export interface InvitationItem {
    id?: string;
    email: string;
    name: string;
    deadline: number;
    role: UserRole;
    editable: boolean;
}